### ISSUE TITLE
Excon::Errors::Error isn't always backwards compatible for Excon::Error

### ISSUE THREAD

**Issue by penman** [excon/excon#580]
Since `Excon::Errors::Error` is a subclass of `Excon::Error`, code like this won't work as expected:
(This example is from the [VCR](https://github.com/vcr/vcr) test suite.)

```ruby
expect {
  Excon.get("http://localhost:#{VCR::SinatraApp.port}/404_not_200", :expects => 200)
}.to raise_error(Excon::Errors::Error)
```

Would there be any problem with changing `Excon::Errors::Error` to be set straight to `Excon::Error`, rather than being a subclass? It would prevent backwards compatibility issues like this.

**Comment by geemus** [comment]
Thanks for the heads up, we are looking into what I suspect is the same issue here: #579 

So, not entirely sure what the solution will be (it may well be exactly as you suggest), but we should hopefully have a backwards compatibility fix and release soonish.

/cc @starbelly maybe this provides some additional context?

**Comment by starbelly** [comment]
@geemus #581 fixes this

**Comment by geemus** [comment]
Should now be fixed and released in 0.51.0. Thanks!
